Provide full signature help with highlighted parameter

MikhailArkhipov opened the pull request:
Fix #70

Ignore Russian translation from pending PR

![image](https://user-images.githubusercontent.com/12820357/33969017-ea18ad16-e01f-11e7-870e-a4d9ead61130.png)

![image](https://user-images.githubusercontent.com/12820357/33969018-f0e488ea-e01f-11e7-9c3e-6c96ef11a111.png)

DonJayamanne commented on src/client/providers/signatureProvider.ts:
```diff
@@ -55,8 +56,13 @@ export class PythonSignatureProvider implements vscode.SignatureHelpProvider {
                 // i.e. line feeds are not respected, long content is stripped.
+                const docLines = def.docstring.splitLines();
+                const label = docLines[0].trim();
+                const documentation = docLines.length > 1 ? docLines.filter((line, index) => index > 0).join(EOL) : '';
```
You could improve the code as follows:
```typescript
const label = docLines.shift();
const documentation = docLines.join(EOL);
```

DonJayamanne commented on src/client/providers/signatureProvider.ts:
```diff
@@ -55,8 +56,13 @@ export class PythonSignatureProvider implements vscode.SignatureHelpProvider {
                 const sig = <vscode.SignatureInformation>{
-                    label: def.description,
+                    label: label,
+                    documentation: documentation,
```
instead of `documentation: documentation,` please use `documentation,`
That's the preferred style for object literals.

DonJayamanne commented on src/client/providers/signatureProvider.ts:
```diff
@@ -65,7 +71,7 @@ export class PythonSignatureProvider implements vscode.SignatureHelpProvider {
                     return <vscode.ParameterInformation>{
                         documentation: arg.docstring.length > 0 ? arg.docstring : arg.description,
-                        label: arg.description.length > 0 ? arg.description : arg.name
+                        label: arg.name
```
Lol, I hang my head in shame for this simple bug. 😞 . Nothing to do here.

DonJayamanne commented:
@MikhailArkhipov looks good. will approve after the release.

MikhailArkhipov commented:
On Python 3.6 VSC gets different signature description which actually does not have signature in it. Need to look into `completion.py`

MikhailArkhipov commented:
@DonJayamanne could you please look one more time. I removed requirement for param name to be a `word` in `completions.py` so we support `...` in signature (ex `print`) and added separate tests for Py3 and Py2 since behavior is different in some functions (notably, `pow` and some others)

DonJayamanne commented on src/test/index.ts:
```diff
@@ -12,7 +12,8 @@ const options: MochaSetupOptions & { retries: number } = {
     timeout: 25000,
-    retries: 3
+    retries: 3,
+    grep: 'Signatures'
```
Please remove this.

DonJayamanne replied:
@MikhailArkhipov please remove the `grep` once done it should be good. I'll approve this now. Just wanted to ensure you got this fixed before merging.
